### DISCUSSION TITLE
Each Array element on a line

### DISCUSSION THREAD

**Discussion by beeb** [toml-rs/toml#782] [Q&A] [answered]
I am modifying an existing document where a key is an Array of strings, and the original document has a line return after the leading `[`, then each element on its own line, indented to the right, then the closing `]` on its own line at the end.

When I get a mutable reference to the key with `doc.get_mut("key").map(|v| v.as_array_mut())` and then replace its content, the format changes and it puts the array on a single line `["item1", "item2"]`.

Is there a way to make it like so?

```
key = [
    "item1",
    "item2",
]
```

**Comment by epage** [comment, marked as answer]
You'll need to set the `Decor` for each element to match your desired style.

As an example
https://github.com/toml-rs/toml/blob/1bfb7d7d79880289fc1f10ef3f6dd0b2b3267c17/crates/toml/src/fmt.rs#L52-L65

**Reply by beeb** [reply]
Very helpful, thanks!
